src/components/BookmarkCard/Details.tsx

```diff
@@ -1,28 +1,73 @@
-import type { PropsWithChildren } from 'react';
+import { PropsWithChildren, useMemo } from 'react';
 import React from 'react';
 import { Link } from '../Link';
 
 import { isEmptyText, stripTags } from '../../lib';
+import classNames from 'classnames';
 
 interface DetailsProps {
     id: string;
     href: string;
     newTab: boolean;
     title?: string;
     description?: string;
+    hasThumbnail: boolean;
+    layout: 'vertical' | 'horizontal';
 }
 
-export function Details({ title, description, href, id, newTab, children }: PropsWithChildren<DetailsProps>) {
+export function Details({
+    title,
+    description,
+    href,
+    id,
+    newTab,
+    layout,
+    hasThumbnail,
+    children,
+}: PropsWithChildren<DetailsProps>) {
+    const { isDescriptionShort, isTitleShort } = useMemo(() => {
```

Comment:
Nitpicking. I feel like `useMemo` isn't necessary here. Is this calculation here really expensive?

Reply:
no, this is not expensive. I used `useMemo` here just for encapsulation
 
Maybe one day js will have that https://github.com/tc39/proposal-do-expressions

Reply:
Just a note: you could use an [IIFE](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression) here fore encapsulation.

Reply:
yeah, i also thought about that, maybe `useMemo` just looks easier to read rather than plenty of brackets
```
    const values= (() => {
       ...
    })();
```
also when somebody will read this code they could miss call at the end and think that `values` is a function

Overall i do not have strong arguments for each of the solutions, `useMemo` just used more often then `IIFE` in our application